apm-sniffer/apm-sdk-plugin/httpclient-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/httpclient/v5/HttpClientDoExecuteInterceptor.java

```diff
@@ -101,9 +101,14 @@ public void handleMethodException(EnhancedInstance objInst, Method method, Objec
         activeSpan.log(t);
     }
 
-    private String getRequestURI(String uri) throws MalformedURLException {
+    private String getRequestURI(String uri) {
```

Comment:
I suggest referring to the implementation of `v3` version and directly using `URI#getPath`

e.g.
```java
private String getRequestURI(URI uri) {
        if (isUrl(uri.toString())) {
            String requestPath = uri.getPath();
            return requestPath != null && requestPath.length() > 0 ? requestPath : "/";
        } else {
            return uri.toString();
        }
    }
```

Reply:
Is this working for you? I thought you are using illegal path, such as two ports. 
This is just to separate URI and URL.

Reply:
I agree with what was said above "span name should be URI rather then URL",
I don't think there is a need to instantiate `URL` here.

The use of urls should only be in this area `Tags.URL.set(span, buildURL(httpHost, uri.toString()));`

Reply:
That works. My point is reminding your original issue only.

Reply:
> I suggest referring to the implementation of `v3` version and directly using `URI#getPath`
> 
> e.g.
> 
> ```java
> private String getRequestURI(URI uri) {
>         if (isUrl(uri.toString())) {
>             String requestPath = uri.getPath();
>             return requestPath != null && requestPath.length() > 0 ? requestPath : "/";
>         } else {
>             return uri.toString();
>         }
>     }
> ```

I didn't find a way to directly get the URI in HttpClient 4.x/5.x plugin.

Reply:
line#58  `String uri = httpRequest.getUri().toString();`

This is originally the `URI`, and in this use case, the `URI#getPath` does not have any exceptions, 
such as:` http://ip:port:port/ ` will be returned is an empty string, why is it necessary to forcibly convert it to a `URL` to obtain the path here?

If it is necessary to handle the incorrect `URL`, it should be done in the `buildURL` method.